CorsixTH/Lua/entities/humanoids/patient.lua

```diff
@@ -298,11 +300,15 @@ end
 --! Returns true if patient agrees to pay for the given treatment.
 --!param disease_id (string): The id of the disease to test
 function Patient:agreesToPay(disease_id)
+  local hosp = self.hospital
   local casebook = self.hospital.disease_casebook[disease_id]
   local price_distortion = self:getPriceDistortion(casebook)
   local is_over_priced = price_distortion > self.hospital.over_priced_threshold
 
-  return not (is_over_priced and math.random(1, 5) == 1)
+  local agreesToPay = not (is_over_priced and math.random(1, 5) == 1)
+  if agreesToPay then self.pay_amount = hosp:getTreatmentPrice(disease_id) end
+
+  return agreesToPay
```

Comment:
```suggestion
  if is_over_priced and math.random(1, 5) == 1 then return false end
  self.pay_amount = hosp:getTreatmentPrice(disease_id)

  return true
```

Does this look better to you? It's probably one of personal preference here. The luadoc at line 300 does already describe what your `return` means; you have just chosen to attach additional meaning with the variable (which isn't always bad).

Reply:
Yes, thanks